app/controllers/medical_records_controller.rb

```diff
@@ -3,9 +3,8 @@ class MedicalRecordsController < ApplicationController
 
   # GET /medical_records or /medical_records.json
   def index
-    @medical_records = MedicalRecord.where(patient_id: params[:patient_id]).all
+    @medical_records = MedicalRecord.where(patient_id: params[:patient_id], usuario_id: current_usuario.id)
```

Comment:
If I'm understanding your relation maybe you can use this way. 

```suggestion
    current_usuario.medical_records.where(patient_id: params[:patient_id])
```

But I'm not sure if this is correct.

Reply:
hmm with instance variable if it works

Reply:
Both works, but is more common use `current_usuario.medical_records`.